service/src/test/java/bio/terra/workspace/pact/ProfileApiTest.java

```diff
@@ -0,0 +1,108 @@
+package bio.terra.workspace.pact;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import au.com.dius.pact.consumer.MockServer;
+import au.com.dius.pact.consumer.dsl.PactDslJsonBody;
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.core.model.RequestResponsePact;
+import au.com.dius.pact.core.model.annotations.Pact;
+import bio.terra.workspace.app.configuration.external.SpendProfileConfiguration;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.spendprofile.SpendProfileId;
+import bio.terra.workspace.service.spendprofile.SpendProfileService;
+import bio.terra.workspace.service.spendprofile.exceptions.SpendUnauthorizedException;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+
+@Tag("pact-test")
+@ExtendWith(PactConsumerTestExt.class)
+public class ProfileApiTest {
+
+  // This will need to be set up with BPM
```

Comment:
I am not sure what you mean about this.

Reply:
outdated comment from before discovering the pathFromProviderState functionality

Reply:
outdated comment from before discovering the pathFromProviderState functionality